Feature/se 43

j0world opened the pull request:
closes SE2-WS23/documentation#43

![E742C22F-199E-4780-A099-FC7165F8FFAC](https://github.com/SE2-WS23/Skillverse-Web/assets/136974808/b224e4a0-5db6-4dea-8a0e-28cf001a23e1)

j0world committed:
Add: quiz page
added quiz page and modified main.jsx to include a link to it

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
fix: implementing recommended changes
making layout more reactive (change from % to vw/vh)
setcurrentQuestion -> setCurrentQuestion
setcurrentAnswer -> setCurrentAnswer
added return to handleAnswerButtonClick()
deleted unnecessary comment handleAnswerButtonClick()
quiztions -> dummyQuestions

j0world committed:
fix: re added LinkedInLearningPage
I think it came here after the merge?
didn't think I'd also have to commit it but here we are :)

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
fix: re added quiz to the router

j0world committed:
fix: added theme data colors
also added empty onClick function for the next button

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
fix: readded quiz to routes

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
Merge branch 'main' into feature/se-43

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,128 @@
+    const [progressValue, setProgressValue] = useState(1);
+
+    const handleAnswerButtonClick = (answer) => {
+        if (dummyQuestions[currentQuestion].correct === answer.target.textContent) // checks if the answer is correct
+            console.log("Correct");
+        else
+            console.log("Incorrect");
```
You consider to show that as visual representation to the user, but I think that is @NicolasKmbr decision

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,128 @@
+            setProgressValue(100); 
+    }
+
+    const nextButtonClick = () => {
```
Can you name this handleNextButtonClick?

j0world replied:
done

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,128 @@
+        return;
+    }
+
+    const dummyQuestions = [ // placeholder for the real questions
```
Remove that comment and please move this mockData to mockData.js

j0world replied:
It has been relocated. 
I also added a random question selection which is now incredibly secure >:)
If that needs to leave again, I would completely understand.

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,128 @@
+                    }}
+                >
+                    
+                    <h1 style={{margin: "0% 0 15vh 0"}}>
```
Why did you use 0% and 15vh in the same statement?

j0world replied:
changed now :)

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,128 @@
+                >
+                    
+                    <h1 style={{margin: "0% 0 15vh 0"}}>
+                        {dummyQuestions[currentQuestion].question}
```
You can consider to also set here a default

j0world committed:
fix: outsourced mock data to mockData.js
added  random question selection
nextButtonClick -> handleNextButtonClick
adjusted margin in line 110

j0world committed:
fix: replaced Math.random() with crypto.getRandomValues()
SonarCloud decided that generating numbers for shuffeling answers with Math.random() is too dangerous

j0world committed:
fix: deleted unnecessary console.log

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
Why can I answer the last question multiple times?

j0world committed:
fix: fixed amount of answers user can give to last question
The user can now only give 1 answer to the last question.

j0world committed:
fix: getting rid of codesmell from prev. commit
line 53: for loop -> for-of loop

j0world committed:
fix: even more code smells
I do not know how to use for-of

j0world committed:
fix: provided default value for dummyQuestions
changed <h1> to Typography in line 128

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -56,7 +57,7 @@ function QuizPage() {
 
-    const dummyQuestions = useMemo(() => randomQuestionsSelection(), []); 
+    const dummyQuestions = useMemo(randomQuestionsSelection, []); 
     const userAnswerArray = useMemo(() => userAnswerArrayLength(), []);
```
Remove this () => like in the line above

j0world committed:
fix: requested changes
line 61: const userAnswerArray = useMemo(() => userAnswerArrayLength(), []); -> const userAnswerArray = useMemo(userAnswerArrayLength, []);
also deleted empty line 129

j0world committed:
fix: used prettier for formatting

Chrissydrx reviewed (approved):
![image](https://github.com/SE2-WS23/Skillverse-Web/assets/37250473/a6ec11ef-8c69-47a8-8851-3a8252b200b5)
